extensions/keycloak-authorization/deployment/src/test/resources/offline-enforcer-pattern-application.properties

```diff
@@ -0,0 +1,18 @@
+# Configuration file
+quarkus.oidc.client-id=backend-service
+quarkus.oidc.credentials.secret=secret
+quarkus.keycloak.devservices.realm-path=quarkus-realm.json
+quarkus.oidc.token.issuer=any
+
+# Enable Policy Enforcement
+quarkus.keycloak.policy-enforcer.enable=true
+quarkus.keycloak.policy-enforcer.lazy-load-paths=false
+
+# Disables policy enforcement for a path
+quarkus.keycloak.policy-enforcer.paths.1.path=/api/public/*
+quarkus.keycloak.policy-enforcer.paths.1.enforcement-mode=DISABLED
+
+quarkus.keycloak.policy-enforcer.paths.2.path=/api/{subfolder}/
+quarkus.keycloak.policy-enforcer.paths.2.enforcement-mode=ENFORCING
```

Comment:
@michalvavrik Did you plan to also test an enforcing mode for `/api/*` ?

Reply:
It's not a problem to test it. I'll add couple of tests in the afternoon. TBH, I didn't do it as I'm not worried (same reason why I didn't add tests for your previous comment prior to the comments), it is not our job to handle this correctly, that is whole point of using Keycloak path matcher - they must provide correct path matching mechanism as they are logic owner. @sberyozkin

Reply:
this PR is only about one thing - using path matcher to resolve path config. `/api/*` enforcing mode is inherent here as it wasn't specified otherwise (need to test it to be sure), isn't it?

Reply:
Added `/api/*` to the `offline-enforcer-pattern-application.properties`, it's already tested when path matcher is invoked with `/api/unknown-url` and behavior didn't change as expected.

Reply:
@michalvavrik I don't really follow all the variations but I wanted to make sure that the code where you check for the pattern does not miss an enforcing mode with a wildcard at the end. You have this code
```
for (PolicyEnforcerConfig.PathConfig pathConfig : getPaths().values()) {
                if (pathConfig.getPath() != null) {

                    if (pathConfig.hasPattern()) {
                        anyPathHasPattern = true;
                        break;
                    }

                    if (pathConfig.getEnforcementMode() == DISABLED) {
                    }
                }
}
```

This code does not catch enforcing mode for `/api/*`. So I'd like to see this case being tested where the same pattern is used for the disabled policy, this is all.
I agree it is a conflict but so it was the case for your original setup which you did to justify the way the above code is done, i.e, to avoid a case where what is meant to be enforced matches the disabled policy path. You had `/api/public/{pattern}` enforced and `/api/public/*` both patterns would match for `/api/public/1`

Reply:
So you want to avoid handling our self situation when there is a wildcard both for enforcing mode and disabled mode. I don't understand why. This part of path matching done by this PR is same as done by Keycloak, so how will it differ?

Reply:
I also agree that ideally it would be handled by Keycloak Authz API, but it can actually be very hard to spot conflicting/overlapping JAX-RS patterns, some of them may have regexpressions, it can be tricky

Reply:
Anyway, now I understand what you proposing, thanks.